airbyte-integrations/connectors/source-amplitude/source_amplitude/components.py

```diff
@@ -0,0 +1,105 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+import gzip
+import io
+import json
+import logging
+import zipfile
+from dataclasses import InitVar, dataclass
+from typing import IO, Any, Iterable, List, Mapping, MutableMapping, Union
+
+import pendulum
+import requests
+from airbyte_cdk.sources.declarative.extractors.record_extractor import RecordExtractor
+from airbyte_cdk.sources.declarative.schema.json_file_schema_loader import JsonFileSchemaLoader
+from airbyte_cdk.sources.declarative.types import Config, Record
+from dataclasses_jsonschema import JsonSchemaMixin
+
+logger = logging.getLogger("airbyte")
+
+
+class AverageSessionLengthRecordExtractor(RecordExtractor):
```

Comment:
can you create issues for all these custom components as described [here](https://docs.google.com/document/d/1LB4gBk0slHj3zNmq1Qtj1U7VkKyTN9eFsVldgQ8T4HQ/edit?usp=drivesdk)?

Reply:
updated